eth_abi/decoding.py

```diff
@@ -594,12 +594,29 @@ def read_data_from_stream(stream):
     def validate_padding_bytes(self, value, padding_bytes):
         pass
 
-    @parse_type_str('string')
+    @parse_type_str('bytes')
     def from_type_str(cls, abi_type, registry):
         return cls()
 
 
-class ByteStringDecoder(StringDecoder):
-    @parse_type_str('bytes')
+class StringDecoder(ByteStringDecoder):
+    @parse_type_str('string')
     def from_type_str(cls, abi_type, registry):
         return cls()
+
+    @staticmethod
+    def decoder_fn(data):
+        try:
+            value = data.decode("utf-8")
+        except UnicodeDecodeError as e:
+            raise UnicodeDecodeError(
```

Comment:
This should probably just be the local `DecodingError` from this library.

Reply:
Also, `raise from e` will keep all the original decoding error details in the stack.